build.assets/Makefile

```diff
@@ -168,25 +170,20 @@ buildbox-centos7:
 		--build-arg LIBBPF_VERSION=$(LIBBPF_VERSION) \
 		--build-arg LIBPCSCLITE_VERSION=$(LIBPCSCLITE_VERSION) \
 		--cache-from $(BUILDBOX_CENTOS7) \
+		$(if $(PUSH),--push) \
 		--tag $(BUILDBOX_CENTOS7) -f Dockerfile-centos7 .
 
 #
 # Builds a Docker buildbox for CentOS 7 FIPS builds
+# Uses regular CentOS 7 buildbox as base, so limited arguments are needed.
 #
 .PHONY:buildbox-centos7-fips
 buildbox-centos7-fips:
```

Comment:
This needs a dependency on `buildbox-centos7` since it needs it to build this buildbox. My build failed because I didn't have it. But when I add that dependency and try `make -C build.assets release-amd64-fips`, I get a different build error:

```
Dockerfile-centos7:5
--------------------
   3 |     # Create an alias to the assets image. Ref: https://github.com/docker/for-mac/issues/2155
   4 |     ARG BUILDBOX_CENTOS7_ASSETS
   5 | >>> FROM ${BUILDBOX_CENTOS7_ASSETS} AS teleport-buildbox-centos7-assets
   6 |
   7 |     ## BASE ###################################################################
--------------------
ERROR: failed to solve: ghcr.io/gravitational/teleport-buildbox-centos7-assets:teleport15: no match for platform in manifest: not found
```

Reply:
I have cleared my image cache and tried again. On this PR's branch, I get the above error. On master I can run:

    make -C build.assets buildbox-centos7

successfully.

Reply:
Actually, it is not successful on master - it just gets a lot further. It seems to be failing on the stuff added in #35624 and #35758 (Build elfutils and zlib with `fPIC` & Fix typo in `Dockerfile-centos7-assets`)

Reply:
@reedloden iterated on this with my testing on amd64, and this is sorted now somewhere amongst all the pushes below here.